app/src/main/java/com/andrew/tasky/presentation/agenda_item_detail/AgendaItemDetailFragment.kt

```diff
@@ -89,32 +92,56 @@ class AgendaItemDetailFragment: Fragment(R.layout.fragment_agenda_item_detail) {
         }
     }
 
+    private fun setupAgendaInitialViewModels() {
+        if (!viewModel.isInitiallySetup.value) {
+            args.agendaItem?.let {
+                viewModel.setTitle(it.title)
+                viewModel.setDescription(it.description)
+                viewModel.setSelectedDate(it.startDateAndTime.toLocalDate())
+                viewModel.setSelectedTime(it.startDateAndTime.toLocalTime())
+                viewModel.setIsDone(it.isDone)
+                viewModel.setSelectedReminderTime(it.reminderTime)
+            }
+            viewModel.setEditMode(args.isInEditMode)
+            viewModel.setInitialSetupToTrue()
+        }
+    }
+
     private fun setupAgendaItemType(){
         binding.apply {
             when (agendaItemType) {
                 AgendaItemType.TASK -> {
-                    taskColorBox.setImageResource(R.drawable.task_icon_box)
-                    taskType.text = getString(R.string.task)
+                    taskColorBox.setImageResource(R.drawable.ic_task_box)
+                    agendaItemTypeTextView.text = getString(R.string.task)
+                    addPhotoLayout.visibility = View.GONE
```

Comment:
If it's just about toggling visible and gone, you can simply use `addPhotoLayout.isVisible = true/false`